lib/cisco_node_utils/bgp.rb

```diff
@@ -122,6 +122,12 @@ def set_args_keys_default
 
     # Attributes:
 
+    # nsr Getter
+    def nsr
+      result = config_get('bgp', 'nsr', @get_args)
```

Comment:
In order to keep Puppet happy, you will need to return 'nil' for cases where it's unsupported. You get this for free with Nexus thanks to the `_exclude` in the YAML, but for XR VRF you need an explicit check:

```ruby
def nsr
  return nil if platform == :ios_xr && @vrf != 'default'
  config_get('bgp', 'nsr', @get_args)
end
```